src/components/QuestionList.tsx

```diff
@@ -0,0 +1,442 @@
+import {
+  ArrowLeftOutlined,
+  DeleteOutlined,
+  LoadingOutlined,
+  SearchOutlined,
+} from '@ant-design/icons';
+import { Button, Divider, Drawer, Input, Rate, Space, Spin } from 'antd';
+import Table, { ColumnsType } from 'antd/lib/table';
+import {
+  ColumnFilterItem,
+  ColumnTitle,
+  FilterConfirmProps,
+  FilterDropdownProps,
+} from 'antd/lib/table/interface';
+import { DataIndex } from 'rc-table/lib/interface';
+import React, {
+  Key,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Link, useParams } from 'react-router-dom';
+
+import {
+  Client,
+  InterviewQuestionModel,
+  QuestionListModel,
+  UpdateQuestionListRequest,
+} from '../services/Client';
+import { getDistinctValues, toPascalCase } from '../utils/stringUtils';
+import styles from './QuestionLists.module.scss';
+
+export default function QuestionLists(): JSX.Element {
+  const { id } = useParams<'id'>();
+  const searchInput = useRef<Input>();
+  const client = useMemo(() => new Client(), []);
+  const [isBeingEdited, setBeingEdited] = useState<boolean>(false);
+  const [list, setList] = useState<QuestionListModel>();
+  useEffect(() => {
+    async function loadList(): Promise<void> {
+      setList(
+        (await client.questionLists(Number(id), undefined, undefined))[0]
+      );
+    }
+    loadList();
+  }, [client]);
+
+  const [allQuestions, setAllQuestions] = useState<InterviewQuestionModel[]>(
+    []
+  );
+  useEffect(() => {
+    async function loadQuestions(): Promise<void> {
+      setAllQuestions(
+        await client.interviewQuestions(undefined, undefined, undefined)
+      );
+    }
+    loadQuestions();
+  }, [client]);
+
+  const [isDrawerVisible, setDrawerVisible] = useState(false);
+  const [questionsToAdd, setQuestionsToAdd] = useState<
+    InterviewQuestionModel[]
+  >([]);
+  const [questionsToRemove, setQuestionsToRemove] = useState<
+    InterviewQuestionModel[]
+  >([]);
+  function setDrawerVisibility(value: boolean | null = null): void {
+    setDrawerVisible((isCurrentlyVisible) => value ?? !isCurrentlyVisible);
+  }
+
+  const [searchText, setSearchText] = useState<React.Key>('');
+
+  const handleSearch = (
+    selectedKeys: React.Key[],
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ): void => {
+    confirm({ closeDropdown: true });
+    setSearchText(selectedKeys[0]);
+  };
+
+  const handleReset = (
+    clearFilters: () => void,
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ): void => {
+    clearFilters();
+    setSearchText('');
+    confirm({ closeDropdown: true });
+  };
+
+  const getColumnProps = (dataIndex: string) => ({
+    title: toPascalCase(dataIndex),
+    dataIndex,
+    key: dataIndex,
+  });
+
+  const getColumnSearchProps = (
+    dataIndex: 'title' | 'category' | 'content' | 'difficulty'
+  ): {
+    title: ColumnTitle<InterviewQuestionModel>;
```

Comment:
I would recommend move into separate type definition. When everything is here it is hard to read where definition and body are defined.